### DISCUSSION TITLE
openUrlInNewTab() doesn't work when record clicked in table

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#10576] [Help] [answered]
I'm able to make the View action open a URL in a new tab with the below code.

```php
Tables\Actions\ViewAction::make()
    ->url(function () {
        return 'https://example.com';
    })
    ->openUrlInNewTab(),
```

However, the entire row is now clickable to that URL and does not open in a new tab. Is there a way to do this or is this a bug?

**Comment by imsyuan** [comment]
I also want to have this feature, is anybody have workaround solution? Thanks.

**Reply by leandrocfe** [reply]
Probably disabling `$table->recordUrl(null)` or `$table->recordAction(null)`

**Reply by intrepidws** [reply]
I can confirm that this works to disable the row from being clickable at all. This somewhat accomplishes my goal, but it feels like there should be a way to allow the row to be clickable _and_ open in a new tab.

**Reply by hproux** [reply]
Try using this trick:
`->recordUrl("https://example.com\" target=\"_blank\"")`

**Reply by intrepidws** [reply]
Neat trick, thanks.

**Reply by rvzug** [reply]
What a hack 👍

**Comment by dagalumin** [comment, marked as answer]
cleaner solution
`$table->openRecordUrlInNewTab();`

**Reply by intrepidws** [reply]
This is much better! Thanks!

**Reply by bilogic** [reply]
but can't specific a `target=`, end up open so many tabs

**Reply by intrepidws** [reply]
Sounds like a good improvement for you to PR.

**Reply by bilogic** [reply]
i think v3 is closed for features and v4 isn't official... so...
